### PR TITLE
NO-JIRA: CPMS Cluster Operator should set related objects

### PR DESCRIPTION
**Pull request by JoelSpeed**
There is an expectation that all cluster operators set at least one namespace and one other object as their related objects. This PR updates our CO generation to set the related objects when they are not as expected.

### CODE REVIEW

File: pkg/controllers/controlplanemachineset/cluster_operator.go [hunk 1]
```diff
@@ -174,3 +205,28 @@ func getStatusForConditionType(conds []configv1.ClusterOperatorStatusCondition,
 
 	return string(configv1.ConditionUnknown)
 }
+
+// relatedObjectsChanged returns true if the related objects on the ClusterOperator have are different from the expected.
```
Comment (damdo) [inline pkg/controllers/controlplanemachineset/cluster_operator.go:209]:
```suggestion
// relatedObjectsChanged returns true if the related objects on the ClusterOperator are different from the expected.
```

**Review by damdo — COMMENTED** [review]
/approve
/lgtm

Only a nit

File: pkg/controllers/controlplanemachineset/cluster_operator.go [hunk 2]
```diff
@@ -24,11 +24,36 @@ import (
...
+	return []configv1.ObjectReference{
+		{
+			Group:    "",
+			Resource: "namespaces",
+			Name:     machineAPINamespace,
+		},
+		{
+			Group:     machinev1.GroupName,
+			Resource:  "controlplanemachinesets",
+			Name:      "cluster",
```
Comment (wking) [inline pkg/controllers/controlplanemachineset/cluster_operator.go:46]: Comparing with [the ones you already set in your ClusterOperator manifest][1], #120 suggests not setting a name here.  Although the `a resource cannot be retrieved by name across all namespaces` being discussed there seems like maybe the problem was not setting `namespace`?  However you go, you probably want the initial-CVO-pre-creation manifest content and this actively-managed-by-your-operator content to pick the same direction.  

[1]: https://github.com/openshift/cluster-control-plane-machine-set-operator/blame/69ab2b776804db8d1baceb26c18cf938498c3f2c/manifests/0000_30_control-plane-machine-set-operator_04_clusteroperator.yaml#L14-L23

**Comment by JoelSpeed** [comment]
Updated now to fix the nit from Dam and to match what we already have in the CVO pre-created manifest

**Comment by damdo** [comment]
LGTM
Will leave to @wking the final label

**Review by wking — APPROVED** [review]
/lgtm

**Pull request merged**